client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/index.tsx

```diff
@@ -22,23 +22,56 @@ import type { SiteSelect } from '@automattic/data-stores';
 
 import './style.scss';
 
-type LaunchpadProps = {
+// This wrapper offloads whether to use siteSlug or siteId to useSite()
+const LaunchpadSiteWrapper: Step = ( {
+	navigation,
+	flow,
+}: {
 	navigation: NavigationControls;
 	flow: string | null;
+} ) => {
+	const site = useSite();
+
+	// Return null while sites are loaded
+	if ( ! site || ! flow ) {
+		return null;
```

Comment:
You are probably aware of this, more of an investigation note to myself: if this component is swapped for the unit test, the tests will fail because it will early return here.

This one is tricky to fix and I get why you made the comment: "Need to have another go at moving the wrapper back into the main component and fixing tests from there."

Currently, the behavior translates to when no siteSlug is provided `http://calypso.localhost:3000/setup/start-writing/launchpad?siteSlug=`, in
stead of redirecting to `/home`, we're stuck here:
![empty blog](https://github.com/Automattic/wp-calypso/assets/6586048/6002aab7-6238-436f-91da-bf9a5ce36234)

Reply:
This wasn't too bad in the end, I was a bit too focused on cleaning up the nulls I think.

Ended up just doing this https://github.com/Automattic/wp-calypso/pull/84532/files?w=1#diff-817fa04f9f4ef10dca8db7c843f23ffa38fea35ddd942b43a34420ed2bec19e1R37-R40

Feel free to improve, I think I'm being overly cautious with the fallbacks I'm pretty sure some of it never gets hit in any react state/context.